AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3148,6 +3148,164 @@ function Debug-AzStorageAccountADObject
     }
 }
 
+function Debug-KerberosTicketEncryption
+{
+    [CmdletBinding()]
+
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Resource group name")]
+        [string]$ResourceGroupName
+    )
+
+    process
+    {
+        $storageAccount = Validate-StorageAccount -ResourceGroupName $ResourceGroupName `
+        -StorageAccountName $StorageAccountName -ErrorAction Stop
+
+        $protocolSettings = (Get-AzStorageFileServiceProperty -StorageAccount $storageAccount -ErrorAction Stop).ProtocolSettings.Smb
+
+        $adObject = Get-AzStorageAccountADObject -StorageAccountName $StorageAccountName `
+            -ResourceGroupName $ResourceGroupName -ErrorAction Stop
+
+        Write-Verbose "Validating the security protocol settings has 'Kerberos' as one of the Smb Authentication Methods"
+
+        $authenticationMethods = $protocolSettings.AuthenticationMethods
+        if ($null -eq $authenticationMethods)
+        {
+            # if null, all types are supported for the storage account
+            $authenticationMethods = "NTLMv2", "Kerberos"
+        }
+        $authenticationMethods = [String]::Join(", ", $authenticationMethods)
+
+        if(!$authenticationMethods.Contains("Kerberos"))
+        {
+            Write-Error -Message "The protocol settings on the storage account does not support 'Kerberos' as one of the Smb Authentication Methods" -ErrorAction Stop
+        }
+
+        Write-Verbose "Validating Kerberos Ticket Encryption setting on the client side is supported"
+        
+        $kerberosTicketEncryptionClient = $adObject.KerberosEncryptionType
+        if(
+            $null -eq $kerberosTicketEncryptionClient -or `
+            0 -eq $kerberosTicketEncryptionClient.Count -or `
+            'None' -eq $kerberosTicketEncryptionClient.Value.ToString()
+            )
+        {
+            Write-Error -Message "No Kerberos Ticket Encryption is supported on the client side" -ErrorAction Stop
+        }

```

Comment:
It's possible that KerberosEncryptionType is not set but a default encryption type is provided by AD. In that case you probably can use klist to get a Keberos ticket and check the Kerberos Encryption Type in the output.

Reply:
Fixed